File: source/classes/Party.java
package source.classes;


public class Party {
	// attributes
	private int number = 0;
	private String name = "";
	private String alias = "";
	private int votes = 0;


	// constructor method
	public Party(String alias) {
		this.alias = alias;
	}


	// methods
	public void displayPartyInfo() {
		System.out.println("Party alias: " + alias);
		System.out.println("Party number: " + number);
		System.out.println("Party name: " + name);
		System.out.println("Party votes: " + votes);
	}

	public int getNumber() {
		return number;
	}
	public void setNumber(int number) {
		this.number = number;
	}

	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}

	public String getAlias() {
		return alias;
	}
	public void setAlias(String alias) {
		this.alias = alias;
	}

	public int getVotes() {
		return votes;
	}
	public void setVotes(int votes) {
		this.votes = votes;
	}
}

File: source/party.cpp
#include "../include/party.h"


Party::Party(const string &alias) {
	this->alias = alias;
}
Party::~Party() {
	//// do nothing
}


void Party::displayPartyInfo() {
	cout << "Party alias: " << alias << endl;
	cout << "Party number: " << number << endl;
	cout << "Party name: " << name << endl;
	cout << "Party votes: " << partyVotes << endl;
	cout << "Total votes: " << totalVotes << endl;
	cout << "\n";
}

const int Party::getNumber() {
	return number;
}
void Party::setNumber(const int number) {
	this->number = number;
}

const string& Party::getName() {
	return name;
}
void Party::setName(const string &name) {
	this->name = name;
}

const string& Party::getAlias() {
	return alias;
}
void Party::setAlias(const string &alias) {
	this->alias = alias;
}

const int Party::getPartyVotes() {
	return partyVotes;
}
void Party::setPartyVotes(const int votes) {
	this->partyVotes = votes;
}

const int Party::getTotalVotes() {
	return totalVotes;
}
void Party::setTotalVotes(const int votes) {
	this->totalVotes = votes;
}

vector<Candidate*> Party::getCandidates() {
	return candidates;
}
void Party::setCandidates(vector<Candidate*> candidates) {
	this->candidates = candidates;
}
void Party::addCandidate(Candidate &candidate) {
	candidates.push_back(&candidate);
}
void Party::removeCandidate(Candidate &candidate) {
	// TODO: erase only the element that matches the candidate
}
